File: LeetCode/Find Pivot Index.ts
function pivotIndex(nums: number[]): number {
  const ps = Array.from(nums);

  for (let i = 1; i < ps.length; i++) {
    ps[i] = ps[i] + ps[i - 1];
  }

  for (let i = 0; i < ps.length; i++) {
    if (ps[i] - nums[i] == ps[ps.length - 1] - ps[i]) return i;
  }

  return -1;
}

File: LeetCode/Find Pivot Index.go
func pivotIndex(nums []int) int {
    ps := make([]int, len(nums));	
	copy(ps, nums);
    
    for i := 1; i < len(nums); i++ {
        ps[i] = ps[i-1] + ps[i]
    }
    
    for i := 0; i < len(nums); i++ {
        if (ps[i] - nums[i] == ps[len(ps)-1] - ps[i]){ return i  }      
    }
    
    return -1
}